dashboard/app/controllers/application_controller.rb

```diff
@@ -231,4 +231,28 @@ def require_levelbuilder_mode
   def require_admin
     authorize! :read, :reports
   end
+
+  # Pairings are stored as an array of user ids in the session
+  # (storing full objects is not a good idea because the session is
+  # saved as a cookie)
+
+  def pairings=(pairings_from_params)
+    return if pairings_from_params.blank?
+
+    session[:pairings] = pairings_from_params.map do |pairing_param|
+      other_user = User.find(pairing_param[:id])
+      if current_user.can_pair_with? other_user
+        other_user.id
+      else
+        # TODO: should this cause an error to be returned to the user?
```

Comment:
nil isn't being .compacted -- the return value of map is -- which is an array that may contain nils (which is the point of compact).

we could thrown an exception instead (which will cause a 500 for the API response and for the error to be logged in honeybadger).